### DISCUSSION TITLE
export history sensore di HA in xls/csv

### DISCUSSION THREAD

**Discussion by poliant** [indomus/forum#110] [HUB: Home Assistant]
ciao,
è possibile esportare la storia di un sensore in xls/csv?

PEr ex. per il sensore indicato sotto, mi servirebbe avere su un file l'andamento così da poterlo graficare a parte e fare troubleshooting incrociando i vari sensori/automazioni

![image](https://user-images.githubusercontent.com/26714536/113846499-d3d87900-9796-11eb-8fc0-1e86c0f400d7.png)

grazie

**Comment by snakuzzo** [comment]
Ti serve per forza in formato da poter usare con Excel ?
Esistono soluzioni ad hoc per fare queste cose...vedi ad esempio [InfluxDB](https://www.home-assistant.io/integrations/influxdb/) e Grafana

**Reply by poliant** [reply]
non conosco né influxDB nè Grafana e pensavo si potesse fare facilmente esportando in xls.

Studio un po' e poi decido... ma nn mi sembra così semplice

**Reply by snakuzzo** [reply]
volendo puoi usare platform [FILE](https://www.home-assistant.io/integrations/file/)...ma personalmente lo ritengo una vera m....da :D

Piuttosto, se proprio ti serve esportarlo in CSV/XLS, al posto tuo creerei uno script python esterno ad HA importando le librerie ad hoc per poter lavorare anche con file excel e/o csv.  Attenzione che con script python esterno NON mi riferisco ai Python Script di HA. 

In linea di massima però, lavorare su xls/csv per fare queste cose credo sia una schifezza...

**Reply by poliant** [reply]
ho provato ad usare file, è anche peggio di quanto dici....

leggendo velocemente su grafana, mi sembra veramente potente e flessibile e direi che vado per questa soluzione (così da averla anche per futuri dashboard)
.... guide da suggerire per installare su docker su raspberrypi 4 e integrarlo in HA?

**Reply by snakuzzo** [reply]
No...ti basta seguire le doc ufficiali (anche su dockerhub) + quelle di HA
